presto-main/src/main/java/com/facebook/presto/operator/scalar/MapSubscriptOperator.java

```diff
@@ -202,6 +206,13 @@ public static Object subscript(boolean legacyMissingKey, MethodHandle keyEqualsM
         if (legacyMissingKey) {
             return null;
         }
-        throw new PrestoException(INVALID_FUNCTION_ARGUMENT, format("%s not present in map", key));
+        throw new PrestoException(INVALID_FUNCTION_ARGUMENT, format("%s not present in map", toObjectValue(keyType, key, session)));
+    }
+
+    private static Object toObjectValue(Type type, Object value, ConnectorSession session)
```

Comment:
I think you want something like PlanPrinter.castToVarchar(...). There's no guarantee that the Object value for the type is printable or that it produces anything meaningful. If the type doesn't support casting to varchar, the message should just say "key not present in map"